src/app/archive/lib/processor.ml

```diff
@@ -684,24 +684,35 @@ module Zkapp_permissions = struct
 
   let table_name = "zkapp_permissions"
 
-  let add_if_doesn't_exist (module Conn : CONNECTION) (perms : Permissions.t) =
+  let add_if_doesn't_exist (module Conn : CONNECTION) (perms : Permissions.t)
+      ~logger =
     let txn_version =
       Mina_numbers.Txn_version.to_int @@ snd perms.set_verification_key
     in
     let%bind versions =
       Protocol_versions.find_txn_version (module Conn) ~transaction:txn_version
     in
-    ( match versions with
-    | Ok [] ->
-        failwith
-          (sprintf "No transaction version exists for the permission, %s"
-             (Permissions.to_yojson perms |> Yojson.Safe.to_string) )
-    | Ok _ ->
-        ()
-    | Error e ->
-        failwith
-          (sprintf "fail to query protocol_versions table, see %s"
-             (Caqti_error.show e) ) ) ;
+    let%bind () =
+      match versions with
+      | Ok [] ->
+          [%log error]
```

Comment:
Shouldn't this continue to fail? I thought the goal was to add the protocol version associated with the genesis block before adding genesis accounts

Reply:
Another thing I could do is to wrap a try-with on add-genesis-account function and capture failures there.